src/api/routes/v1/amendments.ts

```diff
@@ -0,0 +1,315 @@
+ */
+function sortByVersion(
+  prev: AmendmentsInfo | AmendmentInVoting,
+  next: AmendmentsInfo | AmendmentInVoting,
+): number {
+  if (isEarlierVersion(prev.rippled_version, next.rippled_version)) {
+    return 1
+  }
+  return -1
+}
+
+/**
+ * Updates amendments in info cache.
+ *
+ * @returns Void.
+ */
+async function cacheAmendmentsInfo(): Promise<void> {
+  try {
+    cacheInfo.amendments = await query('amendments_info').select('*')
+    cacheInfo.amendments.sort((prev: AmendmentsInfo, next: AmendmentsInfo) =>
+      sortByVersion(prev, next),
+    )
+    cacheInfo.time = Date.now()
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any -- TODO: clean up
+  } catch (err: any) {
+    log.error(err.toString())
+  }
+}
+
+void cacheAmendmentsInfo()
+
+/**
+ * Retrieves amendments enabled on a network.
+ *
+ * @param id - The network id.
+ * @returns List of enabled amendments.
+ */
+async function getEnabledAmendments(id: string): Promise<AmendmentsInfo[]> {
+  const enabled = await query('amendments_enabled')
+    .leftJoin(
+      'amendments_info',
+      'amendments_enabled.amendment_id',
+      'amendments_info.id',
+    )
+    .select(
+      'amendments_enabled.amendment_id AS id',
+      'amendments_info.name',
+      'amendments_info.rippled_version',
+    )
+    .where('amendments_enabled.networks', id)
+
+  enabled.sort((prev: AmendmentsInfo, next: AmendmentsInfo) =>
+    sortByVersion(prev, next),
+  )
+  enabled.forEach((amendment: AmendmentsInfo) => {
+    cacheEnabled.add(amendment.id)
+  })
+
+  return enabled
+}
+/**
+ * Adds a ballot into the amendments mapping.
+ *
+ * @param ballot - Amendment vote from a validator.
+ * @param votingAmendments - Map of amendment to validators vote.
+ */
+function parseAmendmentVote(
+  ballot: BallotAmendmentDb,
+  votingAmendments: {
+    [key: string]: {
+      name: string
+      rippled_version: string | undefined | null
+      validators: Array<{ signing_key: string; ledger_index: string }>
+    }
+  },
+): void {
+  const amendmentsVoted = ballot.amendments ? ballot.amendments.split(',') : []
+  amendmentsVoted.forEach((amendmentId: string) => {
+    if (!(amendmentId in votingAmendments)) {
+      votingAmendments[amendmentId] = {
+        name: '',
+        rippled_version: '',
+        validators: [],
+      }
+    }
+    votingAmendments[amendmentId].validators.push({
+      signing_key: ballot.signing_key,
+      ledger_index: ballot.ledger_index,
+    })
+  })
+}
+
+/**
+ * Retrieves amendments enabled on a network.
+ *
+ * @param id - The network id.
+ * @returns List of enabled amendments.
+ */
+async function getVotingAmendments(id: string): Promise<AmendmentInVoting[]> {
+  const inNetworks: BallotAmendmentDb[] = await query('ballot')
+    .leftJoin('validators', 'ballot.signing_key', 'validators.signing_key')
+    .select('ballot.signing_key', 'ballot.ledger_index', 'ballot.amendments')
+    .where('validators.networks', id)
+
+  const votingAmendments: {
```

Comment:
Why is this a map instead of just a list?

Reply:
I used a map for quicker search when merging with `cacheInfo` and `ballot` data. There might be a more efficient way though but this is what I came up with